### PR TITLE
Basic version of github actions

### PR DESCRIPTION
**Pull request by lukasz126**

### CODE REVIEW

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

File: .github/workflows/ci_basic.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      run: cmake -B ${{env.BUILD_PATH}} -S ${{env.SOURCE}} -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+    - name: Build
+      # Build your program with the given configuration
+      run: cmake --build ${{env.BUILD_PATH}} --config ${{env.BUILD_TYPE}}
+
+    - name: Pre-commit actions
+      uses: pre-commit/action@v3.0.1 
+
+    # TODO uncomment when tests are added
```
Comment (Vicx95) [inline .github/workflows/ci_basic.yaml:38]: please uncomment it and add gtest library with simple test to check that it works

**Review by Vicx95 — CHANGES_REQUESTED** [review]

**Comment by ziobron** [comment]
Thanks @Vicx95 for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Comment by ziobron** [comment]
@lukasz126, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

**Review by Vicx95 — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @Vicx95 for Code Review.
🏅 1 XP granted

**Pull request merged**